src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
+    {
+        $class = new \ReflectionClass(get_called_class());
+
+        return preg_replace('#Plugin$#', '', $class->getShortName());
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return array
+     */
+    final public static function getConfiguration($configuration = array())
```

Comment:
yes that's what I was referring to, let the plugin manager call by itself the getters